.github/CONTRIBUTING.md

```diff
@@ -232,3 +232,58 @@ export const choices: MessageCommandDefinition = {
 ```
 
 > In the above example, the `choices` name is an example and needs to be changed to an appropriate command name.
+
+## Command Permissions
+
+Execution permissions can be configured per-command with each command's `requirements`, which is a `CommandPermissions` object.
+An example `CommandPermissions` object can be seen here:
+```ts
+requirenemts: {
+    permissions?: ['BAN_MEMBERS'],
+    permissionsError?: 'You can\'t do that!',
+    roles?: [Roles.ADMIN, Roles.MODERATOR],
+    rolesError?: 'You\'re not on the team!',
+    rolesBlacklist?: false,
+    channels?: [Channels.GENERAL, Channels.SUPPORT],
+    channelsError?: 'That can\'t be used here!',
+    channelsBlacklist?: true,
+    quietErrors?: false,
+    verboseErrors?: false,
+}
+```
+
+And here is an example command with simple permissions:
+```ts
+import Filter from 'bad-words';
+import { say } from 'cowsay';
+import { CommandDefinition } from '../../lib/command';
+import { Channels, CommandCategory } from '../../constants';
+
+export const cowsay: CommandDefinition = {
+    name: ['cowsay', 'cs'],
+    description: 'Emulates the famous UNIX program `cowsay`.',
+    category: CommandCategory.MEMES,
+    requirements: {
+        channels: [Channels.BOT_COMMANDS],
+        verboseErrors: true,
+    },
+    executor: (msg) => {
+        ...
+    },
+};
+```
+
+### Command Permissions Options
+*NOTE: All fields are optional*
+| Field             | Description |
+| ----------------- | ----------- |
+| permissions       | List of all permissions a member must have. |
+| permissionsError  | Custom error message to return for failed permission checks. If omitted, a generic error will be returned. |
+| roles             | List of `Roles` which a member must have at least one of, or must not have any of if blacklisted. Roles are defined in `constants.ts`. |
+| rolesError        | Custom error message to return for failed role checks. If omitted, a generic error will be returned. |
+| rolesBlacklist    | Boolean flag to indicate whether the list of `roles` should be treated as a blacklist (defaults as whitelist). |
+| channels          | List of `Channels` & `Threads` that the command can be executed in, or is not allowed to be executed in if blacklisted. |
+| channelsError     | Custom error message to return for failed channel checks. If omitted, a generic error will be returned. |
+| channelsBlacklist | Boolean flag to indicate whether the list of `channels` should be treated as a blacklist (defaults as whitelist). |
+| quietErrors       | Boolean flag to disable role messages completely & silently fail out of command execution. Will override all other error message fields. |
+| verboseErrors     | Boolean flag to enable a more verbose error message for failed requirement checks. Any `permissionsError`, `rolesError`, or `channelsError` will override this setting in their respective groups. |
```

Comment:
```suggestion
| Field             | Description                                                                                                                                                                                        |
|:------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| permissions       | List of all Discord type permissions a member must have.                                                                                                                                           |
| permissionsError  | Custom error message to return for a failed permission checks. If omitted, a generic error will be returned.                                                                                       |
| roles             | List of `Roles` which a member must have at least one of, or must not have any of if blacklisted. Roles are defined in `constants.ts`.                                                             |
| rolesError        | Custom error message to return for a failed role checks. If omitted, a generic error will be returned.                                                                                             |
| rolesBlacklist    | Boolean flag to indicate whether the list of `roles` should be treated as a blacklist (defaults as whitelist).                                                                                     |
| channels          | List of `Channels` & `Threads` that the command can be executed in, or is not allowed to be executed in if blacklisted. Channels are defined in `constants.ts`.                                    |
| channelsError     | Custom error message to return for a failed channel checks. If omitted, a generic error will be returned.                                                                                          |
| channelsBlacklist | Boolean flag to indicate whether the list of `channels` should be treated as a blacklist (defaults as whitelist).                                                                                  |
| quietErrors       | Boolean flag to disable permission failure messages completely & silently fail out of command execution. Will override all other error message fields.                                             |
| verboseErrors     | Boolean flag to enable a more verbose error message for failed requirement checks. Any `permissionsError`, `rolesError`, or `channelsError` will override this setting in their respective groups. |

```
Minor edits in text (just personal suggestions). Formatting is just me being a little neurotic (feel free to ignore). And newline at the end of the file.